### ISSUE TITLE
[BUG] Also use native terminal's cursor in menus

### ISSUE THREAD

**Issue by shindere** [anthropics/claude-code#39900]
### Preflight Checklist

- [x] I have searched [existing issues](https://github.com/anthropics/claude-code/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug) and this hasn't been reported yet
- [x] This is a single bug report (please file separate reports for different bugs)
- [x] I am using the latest version of Claude Code

### What's Wrong?

This issue is the continuation of #36765, which has been fixed in 2.1.84.

The native cursor of the terminal is now properly handled in prompts,
but not yet in Claude code's menus. In this menu for instance:

```
 Accessing workspace:

 [PROJECT_DIRECTORY]

 Quick safety check: Is this a project you created or one you trust? (Like your own code, a well-known open source project, or work from your team). If not, take a moment to review what's in this folder first.

 Claude Code'll be able to read, edit, and execute files here.

 Security guide

   1. Yes, I trust this folder
 > 2. No, exit

 Enter to confirm · Esc to cancel
```

Here, like in the prompt case, the cursor sits at the bottom of the screen rather than being on the selected item, where the `>` is.

This is especially interesting in this dialog where the item which is selected by default is not the first one, as one may expect.

### What Should Happen?

The native cursor of the terminal should be on the selected item. Either on the `>` sign, either on the space that follows it, either on the first character of the item itself.

### Error Messages/Logs

```shell

```

### Steps to Reproduce

Simply run claude in a directory where it has not been run before.

### Claude Model

None

### Is this a regression?

No, this never worked

### Claude Code Version

2.1.85 (Claude Code)

### Platform

Anthropic API

### Operating System

Ubuntu/Debian Linux

### Terminal/Shell

Other

### Additional Information

As for #36765, this issue is not OS specific and exists under Linux, Windows and macOS. It affects all the users of assistive technologies, in particular screen readers.

**Comment by qing-ant** [comment]
Preparing a fix for this, and it should be included by Monday's release! Thank you for the reports
